cpp/include/DataStorm/DataStorm.h

```diff
@@ -1829,6 +1833,7 @@ namespace DataStorm
     template<typename Value>
     std::function<std::function<bool(const Value&)>(const std::string&)> makeRegexFilter() noexcept
     {
+        // std::regex_match accepts a const string&.
```

Comment:
Do you mean the `std::regex` constructor? We don't pass criteria to `regex_match`

Reply:
Here is the full code:
```cpp
        // std::regex_match accepts a const string&.
        return [](const std::string& criteria)
        {
            std::regex expr(criteria);
            return [expr](const Value& value)
            {
                std::ostringstream os;
                os << value;
                return std::regex_match(os.str(), expr);
            };
        };
```

Reply:
Value can be anything that defines `<<` osstream operator, it doesn't have to be a string.

For the regex constructor can we use 8 https://en.cppreference.com/w/cpp/regex/basic_regex/basic_regex

And do:

```
return [](std::string criteria)
        {
            std::regex expr(std::move(criteria));
```

I think the complication here is this template factory are used for different things and, pass by value might be not supported for all usages.

Reply:
What I meant with this comment is a typical regex filter function - regex_match - accepts a `const string&` and not a `string_view` (which in theory it could), and as a result it's ok/reasonable for our makeRegexFilter function to return a function that accepts a `const string&` and not a `string_view`.